src/utils/version_detector.py

```
@@ -0,0 +1,281 @@
+    for pattern in patterns:
+        match = re.search(pattern, file_path, re.IGNORECASE)
+        if match:
+            version = match.group(1)
+            # Normalize version format
+            if version.lower() in ["latest", "current", "stable"]:
+                return version.lower()
+            else:
+                # Preserve R prefix for Check Point versions
+                if version.lower().startswith("r"):
+                    return "R" + version[1:]  # Normalize to uppercase R
+                return version
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)